src/test/kotlin/org/rust/ide/intentions/UnwrapSingleExprIntentionTest.kt

```diff
@@ -0,0 +1,118 @@
+package org.rust.ide.intentions
+
+import org.rust.lang.RustFileType
+import org.rust.lang.RustTestCaseBase
+
+class UnwrapSingleExprIntentionTest : RustTestCaseBase() {
+    override val dataPath = ""
+
+    fun testAvailableLambdaUnwrapBracesSingleExpression() = doTest(
+        """
+        fn main() {
+            {
+                <caret>42
+            }
+        }
+        """
+        ,
+        """
+        fn main() {
+            42
+        }
+        """
+    )
+
+    fun testAvailableLambdaUnwrapBraces() = doTest(
+        """
+        fn main() {
+            |x| { x *<caret> x }
+        }
+        """
+        ,
+        """
+        fn main() {
+            |x| x * x
+        }
+        """
+    )
+
+    fun testAvailableUnwrapBracesSingleExpressionIf() = doTest(
```

Comment:
I think that for cases where the intention should not be available it should be possible to specify only a single input string.